core/common/src/main/java/alluxio/client/file/PrestoContext.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file;
+
+import alluxio.client.quota.CacheQuota;
+import alluxio.client.quota.CacheScope;
+import alluxio.wire.FileInfo;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * Context when running in Presto.
```

Comment:
@apc999 quotas will only be supported for Presto?

Reply:
@apc999 I also have the same question.  Is this class specific for Presto? Is it better to make this class generic and add a Presto specific subclass in the Presto repo?

Reply:
I think both quota and scope are generic fields, which would be helpful to presto, spark-sql and etc..
So we could rename this class to CacheContext, can we?

Reply:
@ggezer @yuanzhanhku @beinan  these two fields are only added to support Presto previously. Note that all the other URIStatus field are retrieved from Alluxio master but these two are only set in Presto code https://github.com/prestodb/presto/blob/master/presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java#L90-L92

I also agree that these are generic field but unfortunately they are not currently something supported generally by Alluxio. So I prefer to group these Presto-specific logic into PrestoContext to have better management. We can always move them out like other fields once they are supported

Reply:
> @apc999 I also have the same question. Is this class specific for Presto? Is it better to make this class generic and add a Presto specific subclass in the Presto repo?

I will create a subclass of this one in Presto-side, e.g., storing `HiveFileContext` and extend `incrementCounter`